### PR TITLE
Document Identity System changes for Web Chat

### PR DESCRIPTION
**Pull request by hs-saunved**
Documents all the JS APIs, events, and other details required for integrating the new Identity system.

### CODE REVIEW

**Review by hs-umang — APPROVED** [review]

File: docs/web-chat/api.mdx [hunk 1]
```diff
@@ -137,6 +137,151 @@ You should call this API before a conversation starts. In other words, it works
 
 </Admonition>
 
+## Login {#login}
+
```
Comment (KrishnaMB-PM) [inline docs/web-chat/api.mdx:141]: In this new section, we need to mention that it is in Beta and to enable touch base with Customer Success. Mention it as a note separately.

File: docs/web-chat/api.mdx [hunk 2]
```diff
@@ -137,6 +137,151 @@ You should call this API before a conversation starts. In other words, it works
...
+     last_city?: "string",
+     age?: "string",
+     lifetime_value?: "string",
+     user_persona?: "string",
+     user_vip_segment?: "string",
+     user_support_status?: "string"
+     last_active_date?: "string" // epoch time in ms/seconds
+     accepted_t_and_c?: "string" // true/false converted to string
+     preferred_language?:"string"
+     full_privacy_enabled?: "string"
```
Comment (KrishnaMB-PM) [inline docs/web-chat/api.mdx:177]: In the Login API, Full Privacy Flag is present, are we allowing to modify from global fields as well?

File: docs/web-chat/api.mdx [hunk 3]
```diff
@@ -137,6 +137,151 @@ You should call this API before a conversation starts. In other words, it works
...
+
+## Logout {#logout}
+
+The Logout API is introduced along with the [new Identity system](/web-chat/users/#managing-users-with-the-identity-system).
+You can refer to the linked section for details.
+
+
+## Update global user data {#update-global-user-data}
+
+This event is used in conjunction with the [new Identity system](/web-chat/users/#managing-users-with-the-identity-system) and is used to update the global data associated with a logged-in user.
```
Comment (KrishnaMB-PM) [inline docs/web-chat/api.mdx:153]: Link the KB article related to fields when published. Not needed right now.

File: docs/web-chat/api.mdx [hunk 4]
```diff
@@ -137,6 +137,151 @@ You should call this API before a conversation starts. In other words, it works
...
+You can plug this API call into any place where you collect user data. However, be careful not to call it in any
+useEffect-like callbacks or frequently-firing events, since rate-limiting restrictions may kick in.
+
+App user data is also referred to as **app attributes**.
+
+</Admonition>
+
+For a list of errors that can occur when using this API, refer [UPDATE_APP_USER_DATA_FAILED](#update-app-user-data-failed).
+
+## Add user identities {#add-user-identities}
```
Comment (KrishnaMB-PM) [inline docs/web-chat/api.mdx:258]: Let us make "Add user identities" the first thing even before the User Profile related information? Happy to discuss if I am missing context on the reason for adding it after the user profile information.

File: docs/web-chat/users.mdx [hunk 5]
```diff
@@ -173,3 +183,228 @@ When making a request to Helpshift with no `userAuthToken` or an invalid `userAu
...
 [How do I test that I have User Identity Verification set up correctly?](https://support.helpshift.com/hc/en/13-helpshift-technical-support/faq/881-user-identity-verification-how-do-i-test-that-i-have-user-identity-verification-set-up-correctly/)
+
+## Managing users with the Identity System {#managing-users-with-the-identity-system}
+
+Setting up the Identity System requires some development effort, but provides many advantages over the legacy
+method in the form of faster context collection, easier agent experience, and enhanced security and spam protection.
+
+### Prerequisites
+
+1. Ensure that the Identity System is enabled for your domain (reach out to Helpshift support if you wish to enable this)
```
Comment (KrishnaMB-PM) [inline docs/web-chat/users.mdx:194]: Reach out to Helpshift Customer Success.

File: docs/web-chat/api.mdx [hunk 6]
```diff
@@ -137,6 +137,151 @@ You should call this API before a conversation starts. In other words, it works
 
 </Admonition>
 
+## Login {#login}
```
Comment (KrishnaMB-PM) [inline docs/web-chat/api.mdx:140]: Can we add all the new identity system related pieces in one single place for easy reference similar to SDK dev docs. Customers might find it very difficult like this.?

File: docs/web-chat/users.mdx [hunk 7]
```diff
@@ -173,3 +183,228 @@ When making a request to Helpshift with no `userAuthToken` or an invalid `userAu
...
+is invalidated before 24 hours elapse.
+
+### Troubleshooting and tips {#troubleshooting-and-tips}
+
+Login failure can be detected by listening to the [`onWebChatRuntimeError` event](/web-chat/api/#on-webchat-runtime-error).
+The possible reasons for login failure are cited in the [`LOGIN_FAILED` runtime errors](/web-chat/api/#login-failed).
+
+If your login is still failing, you can investigate these other potential reasons:
+1. Ensure that your JWT endpoint is generating tokens in the accepted format
+2. Ensure that your JWT endpoint is signing tokens with the shared secret
```
Comment (KrishnaMB-PM) [inline docs/web-chat/users.mdx:408]: Should we link to the KB article of the secret key generation? This article: https://helpshift.atlassian.net/wiki/spaces/~7120201150c255595f403cad6652223bcf8661/pages/4605083915/User+Hub+User+Data+Security

File: docs/web-chat/identity-users.mdx [hunk 8]
```diff
@@ -0,0 +1,697 @@
+---
+sidebar_position: 45
+title: User Hub
```
Comment (hs-umang) [inline docs/web-chat/identity-users.mdx:3]: We are using `Identity User Management` in SDK X dev docs. We should keep it consistent in both SDK X & web chat dev docs.
https://developers.helpshift.com/sdkx_ios/identity-users/

**Review by KrishnaMB-PM — APPROVED** [review]
Good to go to production.

**Pull request merged**